src/Sentry/Protocol/Builders/EnvelopeItemBuilder.cs

```diff
@@ -24,18 +25,25 @@ public EnvelopeItemBuilder AddHeader(string key, object value)
         }
 
         /// <summary>
-        /// Sets the payload data to the specified byte array.
+        /// Sets the payload data to the specified stream.
         /// </summary>
-        public EnvelopeItemBuilder SetData(byte[] data)
+        public EnvelopeItemBuilder SetStream(Stream stream)
         {
-            _data = data;
+            _stream = stream;
             return this;
         }
 
+        /// <summary>
+        /// Sets the payload data to the specified byte array.
+        /// </summary>
+        public EnvelopeItemBuilder SetStream(byte[] data) => SetStream(
+            new MemoryStream(data)
+        );
+
         /// <summary>
         /// Sets the payload data to the specified string.
         /// </summary>
-        public EnvelopeItemBuilder SetData(string data) => SetData(
+        public EnvelopeItemBuilder SetStream(string data) => SetStream(
```

Comment:
Should be possible I think if `SentryEvent` inherits from `EnvelopeItem` (or maybe another common abstraction). That way we should be able to place events directly into envelopes without first serializing them.

Reply:
The exact same discussion is going on right now on Java:
https://github.com/getsentry/sentry-java/pull/971

Maybe we can align somehow. At least brainstorm together.

There's an interface there (which I'm trying to get rid of if we find a way to)

Reply:
we could get an interface, or basically envelope items could have something like `Func<Stream>` or whatever. So we can easily create:

`new EnveloepItem(() => File.OpenRead("file"))`.

Reply:
This is already possible out of the box (i.e. `new EnvelopePayload(File.OpenRead(...))`), the challenge is incorporating an abstract structure instead of an actual stream (in this case `SentryEvent`).
I think the best solution is to add `IEnvelopePayload` and then have `StreamEnvelopePayload` or `EventEnvelopePayload` etc.